sdk/keyvault/keyvault-secrets/tests/CRUD.tests.ts

```diff
@@ -0,0 +1,280 @@
+      clientId,
+      clientSecret,
+      tenantId,
+      {
+        tokenAudience: "https://vault.azure.net"
+      }
+    );
+
+    client = new SecretsClient(url, credential);
+    version = "";
+  });
+
+  it("can add a secret", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    after(deleteSecretAfter(secretName));
+    const result = await client.setSecret(secretName, secretValue);
+
+    assert.equal(result.name, secretName, "Unexpected secret name in result from setSecret().");
+    assert.equal(result.value, secretValue, "Unexpected secret value in result from setSecret().");
+  });
+
+  it("cannot create a key with an empty name", async () => {
+    const secretName = "";
+    const secretValue = getUniqueName("value");
+    let error;
+    try {
+      await client.setSecret(secretName, secretValue);
+      throw Error("Expecting an error but not catching one.");
+    } catch (e) {
+      error = e;
+    }
+    assert.equal(
+      error.message,
+      `"secretName" with value "" should satisfy the constraint "Pattern": /^[0-9a-zA-Z-]+$/.`,
+      "Unexpected error while running setSecret with an empty string as the name."
+    );
+  });
+
+  it("can set a secret with Empty Value", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = "";
+    after(deleteSecretAfter(secretName));
+    const result = await client.setSecret(secretName, secretValue);
+
+    assert.equal(result.name, secretName, "Unexpected secret name in result from setSecret().");
+    assert.equal(result.value, secretValue, "Unexpected secret value in result from setSecret().");
+  });
+
+  it("cannot create a key with a null name", async () => {
+    const secretName = null;
+    let error;
+    try {
+      await client.setSecret(secretName, "");
+      throw Error("Expecting an error but not catching one.");
+    } catch (e) {
+      error = e;
+    }
+    assert.equal(
+      error.message,
+      "secretName cannot be null or undefined.",
+      "Unexpected error while running setSecret with an empty string as the name."
+    );
+  });
+
+  it("can set a secret with attributes", async () => {
+    const secretName = getUniqueName("secret");
+    const secretValue = getUniqueName("value");
+    const expiryDate = new Date("3000-01-01");
+    after(deleteSecretAfter(secretName));
+    await client.setSecret(secretName, secretValue, { expires: expiryDate });
+
+    const updated = await client.getSecret(secretName);
+
+    // TODO: Investigate. The service seems to change the milliseconds part of the expiry date.
+    // For now just compare year, month, and date in assertion.
+    assert.ok(
+      updated.expires!.getUTCFullYear() === expiryDate.getUTCFullYear() &&
+        updated.expires!.getUTCMonth() === expiryDate.getUTCMonth() &&
+        updated.expires!.getUTCDate() === expiryDate.getUTCDate(),
+      "Expect attribute 'expires' to be updated."
```

Comment:
Is it right to call this updated here (and on line 103 above)? The secret seems to be set but not updated in this test.